c7n/resources/ami.py

```diff
@@ -199,6 +287,143 @@ def process_image(self, client, image):
+                resource: ami
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - '112233445566'
+                      - 'arn:aws:organizations::112233445566:organization/o-xxyyzzaabb'
+                      - 'arn:aws:organizations::112233445566:ou/o-xxyyzzaabb/ou-xxyy-aabbccdd'
+                actions:
+                  - type: set-permissions
+                    remove: matched
+                # To remove all permissions
+                # - type: set-permissions
+                # To remove public permissions
+                # - type: set-permissions
+                #   remove:
+                #     - all
+                # To remove specific permissions
+                # - type: set-permissions
+                #   remove:
+                #     - '223344556677'
+                #     - 'arn:aws:organizations::112233445566:organization/o-zzyyxxbbaa'
+                #     - 'arn:aws:organizations::112233445566:ou/o-zzyyxxbbaa/ou-xxyy-ddccbbaa'
+                # To set specific permissions
+                # - type: set-permissions
+                #   remove: matched
+                #   add:
+                #     - '223344556677'
+                #     - 'arn:aws:organizations::112233445566:organization/o-zzyyxxbbaa'
+                #     - 'arn:aws:organizations::112233445566:ou/o-zzyyxxbbaa/ou-xxyy-ddccbbaa'
+    """
+
+    schema = type_schema(
+        'set-permissions',
+        remove={'oneOf': [
+            {'enum': ['matched']},
+            {'type': 'array', 'items': {'type': 'string'}}
+        ]},
+        add={'type': 'array', 'items': {'type': 'string'}}
+    )
+
+    permissions = ('ec2:ResetImageAttribute', 'ec2:ModifyImageAttribute',)
+
+    def validate(self):
+        if self.data.get('remove') == 'matched':
+            found = False
+            for f in self.manager.iter_filters():
+                if isinstance(f, AmiCrossAccountFilter):
+                    found = True
+                    break
+            if not found:
+                raise PolicyValidationError(
+                    "policy:%s filter:%s with matched requires cross-account filter" % (
+                        self.manager.ctx.policy.name, self.type))
+
+    def process(self, images):
+        client = local_session(self.manager.session_factory).client('ec2')
+        for i in images:
+            self.process_image(client, i)
+
+    def process_image(self, client, image):
+        to_add = self.data.get('add')
+        to_remove = self.data.get('remove')
+        # Default is to remove all permissions
+        if not to_add and not to_remove:
+            return client.reset_image_attribute(
+                ImageId=image['ImageId'], Attribute="launchPermission")
+        remove = []
+        add = []
+        account_regex = re.compile('\\d{12}')
+        org_regex = re.compile('arn:[a-zA-Z]+:organizations::\\d{12}:organization/o-.*')
```

Comment:
This will die on non-standard AWS partitions that include the dash character (e.g. `aws-cn`, `aws-us-gov`). One alternative would be tacking that dash to the end (`[a-zA-Z-]`) and another might be just looking for any non-colons before the org bit (`[^:]+`).

Reply:
Good catch - will fix.